### PR TITLE
df-98->Perform query validation at the df-server level for addToStore

### PR DESCRIPTION
**Pull request by vijaypoojari**
This PR includes:
- Query Validation for addToStore, getFromStore, UpdateInStore and DeleteFromStore.
- It checks the input parameters against the set of rules for 1311 and 1411 witsml doc.
- if the CRUD operation is successful it passes 1, else it checks the params and pass the proper witsml 
   response code that is encountered first.
- The validation of parameters in performed in validation class and it used enums to return the proper 
   code.
- QueryValidation class passes the parameters to validation class to perform operation.
- ValidationResult returns the boolean for the enum.

This connects to #98 
This connect to #174

### CODE REVIEW

**Comment by vijaypoojari** [comment]
@cherrera2001  ok i will make the changes

**Comment by randypitcherii** [comment]
@vijaypoojari 	please merge master into this. May have some conflicts

**Comment by niaalex** [comment]
@vijaypoojari please do the needed to get this item updated asap; are you waiting on anything?? please confirm

File: df-server/.eclipse-pmd [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (randypitcherii) [reply df-server/.eclipse-pmd]: @vijaypoojari 	please remove this file from the commit and add `**/.eclipse-pmd` to the project .gitignore file.

[the remark this answers is not in the document]
Comment (vijaypoojari) [reply df-server/.eclipse-pmd]: removed eclipse-pmd from df-server.

**Comment by randypitcherii** [comment]
Hey @vijaypoojari, please move the validation checks to the top of the store methods. We want to check validity before building queries.

**Comment by vijaypoojari** [comment]
@randypitcherii Sure will move the QueryChecks above store methods.

**Comment by cherrera2001** [comment]
@vijaypoojari this needs to be refactored a bit

1) Fix getXMLDocument so that it actually works
2) you need to make it so that you only create the dom once...its extremely inefficent to create the dom repeatedly.

**Comment by vijaypoojari** [comment]
@cherrera2001 sure i will also work on them.

**Comment by vijaypoojari** [comment]
Some methods for validation checks for UID still uses DOM, i am working on them.

**Comment by vijaypoojari** [comment]
@cherrera2001  for UID check the method is checkNotNullUid, the same approach will go for UOM

**Pull request closed without merging**